frontend/src/app/dataset/dataset.component.ts

```diff
@@ -342,7 +342,18 @@ export class DatasetComponent {
     this.normalizationY = !this.normalizationY;
     for (const i in this.sample.data) {
       const new_data = new Array<number>(this.sample.data[0]['arr'].length);
-      this.normalizeYHelper(i, !this.normalizationY, new_data);
+      const [min, max] = this.sample.data[i]['minmax'];
+      // Edge case where a straight line trace would cause a divide by 0.
+      // If a trace with all 0s occurs, it is not normalized in normalizeYHelper.
+      if (min !== 0 && min === max) {
```

Comment:
I would put this part in `normalizeYHelper` as well. Currently this fragments the part of the code.